HotFix.Specification/Specification.cs

```diff
@@ -47,21 +48,29 @@ public Specification Verify(Action<Session, IConfiguration> verification)
         public Specification Run()
         {
             var clock = new VirtualClock();
-            var transport = new VirtualTransport(clock, Instructions);
+            var transport = new VirtualTransport(clock, Instructions, () => LogoutRequested = true);
             var engine = new Engine { Clocks = c => clock, Transports = c => transport };
 
             var session = (Session) null;
 
             try
             {
-                session = engine.Open(Configuration);
+                using (session = engine.Open(Configuration))
+                {
+                    session.Logon();
 
-                session.Logon();
+                    while (session.Active && !LogoutRequested)
```

Comment:
Calling `Logout` will set `Active` to `false`, so that it doesn't get called again when exiting the `while(Active)` loop, and to also escape the loop. Getting a logout from the other side sets `Active` to `false` in order to escape the loop, but never sends a logout back because `Logout` does not want to be called again if `Active` is false. So logging out at the other side's request doesn't work properly. We could introduce more flags, but that might get more complicated.

Another, small, issue is that transport is needed in order to create the session, so adding the reverse dependency might make things a bit too difficult to follow when reading the setup? Not sure.